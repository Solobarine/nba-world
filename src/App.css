* {
  margin: 0;
  padding: 0;
  color: #fff;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

body {
  width: 480px;
  overflow-x: hidden;
  height: 100vh;
  background-color: #fc5193;
}

.App {
  background-color: #fc5294;
}

.network {
  width: 15%;
  display: flex;
  justify-content: space-between;
}

.cell-info {
  display: flex;
  justify-content: space-between;
  color: #fff;
  font-size: 17px;
  padding: 12px;
  background-color: #ec407a;
}

.icons {
  width: 8%;
  display: flex;
  justify-content: space-between;
}

.nba-section {
  background-color: #fc5193;
  height: 80px;
  text-align: center;
  padding: 45px;
}

.category {
  display: flex;
  width: 480px;
  flex-wrap: wrap;
}

.category > div {
  width: 49%;
  background-color: #da4580;
  height: 220px;
  border: 1px solid #555;
  border-radius: 10px;
}

.category > div:nth-child(4n+0),
.category > div:nth-child(4n+1){
  background-color: #fa4580;
}

.category > div i {
  font-size: 20px;
  text-align: right;
  margin-top: 30px;
  margin-right: 25px;
  color: #fff;
}

.category > div .c-title {
  text-align: center;
  margin-top: 70px;
  font-size: 16px;
}

.header {
  display: flex;
  justify-content: space-between;
  padding: 12px;
  background-color: #ec407a;
}

.teams > div {
  width: 90%;
  display: flex;
  background-color: #ce4177;
  padding: 15px;
  border: 1px solid #ff6964;
  margin-bottom: 10px;
  margin-left: 10px;
}

.teams > div p {
  width: 33.33%;
  text-align: center;
  font-size: 22px;
}

.team:nth-child(odd) {
  background-color: #ec470a;
}

.atlanticPage div .fa-arrow-right-long {
  display: none;
}

.atlanticPage div .c-title {
  font-size: 18px;
  text-align: center;
  margin-top: 30px;
}

.atlanticPage .atlantic {
  height: 80px;
}

.centralPage div .fa-arrow-right-long {
  display: none;
}

.centralPage .central {
  height: 80px;
}

.centralPage div .c-title {
  font-size: 18px;
  text-align: center;
  margin-top: 30px;
}

.northWestPage div .fa-arrow-right-long {
  display: none;
}

.northWestPage div .c-title {
  font-size: 18px;
  text-align: center;
  margin-top: 30px;
}

.northWestPage .northwest {
  height: 80px;
}

.pacificPage div .fa-arrow-right-long {
  display: none;
}

.pacificPage div .c-title {
  font-size: 18px;
  text-align: center;
  margin-top: 30px;
}

.pacificPage .pacific {
  height: 80px;
}

.southEastPage div .fa-arrow-right-long {
  display: none;
}

.southEastPage div .c-title {
  font-size: 18px;
  text-align: center;
  margin-top: 30px;
}

.southEastPage .southeast {
  height: 80px;
}

.southWestPage div .fa-arrow-right-long {
  display: none;
}

.southWestPage div .c-title {
  font-size: 18px;
  text-align: center;
  margin-top: 30px;
}

.southWestPage .southwest {
  height: 80px;
}
