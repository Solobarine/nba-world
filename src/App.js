import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Routes, Route } from 'react-router-dom';
import CellInfo from './components/CellInfo';
import { getTeams } from './features/nba/nbaReducer';
import Home from './pages/Home';
import AllNbaTeams from './pages/AllNbaTeams';
import AtlanticPage from './pages/AtlanticPage';
import CentralPage from './pages/CentralPage';
import NorthWestPage from './pages/NorthWestPage';
import PacificPage from './pages/PacificPage';
import SouthEastPage from './pages/SouthEastPage';
import SouthWestPage from './pages/SouthWestPage';

const App = () => {
  const teams = useSelector((store) => store);
  const status = useSelector((store) => store.status);
  const dispatch = useDispatch();
  console.log(teams, status)

  useEffect(() => {
    dispatch(getTeams);
  }, []);
  const showTeams = () => {
    if (status === 'pending') {
      return <div className="loading">Loading, Please Wait...</div>
    }
    if (status === 'rejected') {
      return <div className="rejected">Sorry, We could not get the teams.</div>
    } 
  }
  return (
    <div className="App">
      <CellInfo />
      <Routes>
        <Route path="/" element={<Home/>} />
        <Route path="/home/all" element={<AllNbaTeams/>} />
        <Route path="/home/atlantic" element={<AtlanticPage/>} />
        <Route path="/home/central" element={<CentralPage/>} />
        <Route path="/home/northwest" element={<NorthWestPage/>} />
        <Route path="/home/pacific" element={<PacificPage/>} />
        <Route path="/home/southeast" element={<SouthEastPage/>} />
        <Route path="/home/southwest" element={<SouthWestPage/>} />
      </Routes>
    </div>
  );
}

export default App;
