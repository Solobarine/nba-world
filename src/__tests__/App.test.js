import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../features/configureStore';

describe('App', () => {
  test('Check if App renders correctly', () => {
    const app = renderer.create(<Provider store={store}><BrowserRouter><App /></BrowserRouter></Provider>).toJSON(); // eslint-disable-line
    expect(app).toMatchSnapshot();
  });
});
