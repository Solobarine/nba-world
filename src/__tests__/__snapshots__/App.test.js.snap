// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Check if App renders correctly 1`] = `
<div
  className="App"
>
  <section
    className="cell-info"
  >
    <div
      className="network"
    >
      <i
        className="fa-solid fa-ellipsis"
      />
      <p
        className="carrier"
      >
        VIA
      </p>
      <i
        className="fa-solid fa-wifi"
      />
    </div>
    <p
      className="date"
    >
      00:00:00
    </p>
    <div
      className="icons"
    >
      <i
        className="fa-brands fa-bluetooth"
      />
      <i
        className="fa-solid fa-battery-full"
      />
    </div>
  </section>
  <div
    className="home-section"
  >
    <div
      className="nba-section"
    >
      <img
        alt=""
        className="logo"
        src="#"
      />
      <h2
        className="title"
      >
        NBA
      </h2>
    </div>
    <div
      className="category"
    >
      <div
        className="atlantic"
      >
        <a
          href="/home/atlantic"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-arrow-right-long"
          />
        </a>
        <h2
          className="c-title"
        >
          Atlantic Division Teams
        </h2>
      </div>
      <div
        className="central"
      >
        <a
          href="/home/central"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-arrow-right-long"
          />
        </a>
        <h2
          className="c-title"
        >
          Central Division Teams
        </h2>
      </div>
      <div
        className="northwest"
      >
        <a
          href="/home/northwest"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-arrow-right-long"
          />
        </a>
        <h2
          className="c-title"
        >
          NorthWest Division Teams
        </h2>
      </div>
      <div
        className="pacific"
      >
        <a
          href="/home/pacific"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-arrow-right-long"
          />
        </a>
        <h2
          className="c-title"
        >
          Pacific Division Teams
        </h2>
      </div>
      <div
        className="southeast"
      >
        <a
          href="/home/southeast"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-arrow-right-long"
          />
        </a>
        <h2
          className="c-title"
        >
          SouthEast Division Teams
        </h2>
      </div>
      <div
        className="southwest"
      >
        <a
          href="/home/southwest"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-arrow-right-long"
          />
        </a>
        <h2
          className="c-title"
        >
          SouthWest Division Teams
        </h2>
      </div>
    </div>
  </div>
</div>
`;
